packages/relay/src/lib/types/debug.ts

```diff
@@ -73,5 +73,5 @@ export interface TraceBlockByNumberTxResult {
   /**
    * The result of the trace, which can be either a {@link CallTracerResult} or an {@link EntityTraceStateMap}.
    */
-  result: CallTracerResult | EntityTraceStateMap;
+  result: CallTracerResult | EntityTraceStateMap | null;
```

Comment:
Do we have a link of which spec we are following? There is no [official spec](https://ethereum.github.io/execution-apis/api-documentation) for these `debug_*` methods. So we probably want to make it clear which spec we are following.

For example, to answer things like "why is `null` a valid return value here?".

Reply:
Yes, totally. As discussed offline, it's not easy to find a proper spec for these `debug_*` endpoints. QuickNode includes them in their free tier, so we can try them out, but not Alchemy for example.

I think for now returning `null` is fine, but we'd need to find a better spec to adhere to. For example, in the case of "transaction not found", Anvil (Foundry's development EVM) returns an "empty" result (instead of `null`)

```
  "result": {
    "failed": false,
    "gas": 0,
    "returnValue": "",
    "structLogs": []
  }
```

It would be interesting to see what `geth` or Hardhat return in this case as well. With this information we might be able to have a pseudo-spec to adhere to.